src/lib.rs

```diff
@@ -133,6 +133,12 @@ pub trait CustomizeConnection<C, E>: fmt::Debug + Send + Sync + 'static {
     fn on_acquire(&self, conn: &mut C) -> Result<(), E> {
         Ok(())
     }
+
+    /// Called with connections before they are dropped from the pool
+    ///
+    /// The default implementation does nothing
+    #[allow(unused_variables)]
+    fn on_release(&self, conn: &mut C) {}
```

Comment:
Kind of. `drop_conn()` can accept a `Vec` but it will also have to take the `internals` lock by value. I'll send an updated pull request later today.